k8s_attach/test/Tiltfile

```diff
@@ -0,0 +1,15 @@
+load('../Tiltfile', 'k8s_attach')
+
+# NB: attach() is intended for attaching to existing
+# deployments. Normally, you'd do the kubectl apply
+# outside of the main
+local('kubectl apply -f deployment.yaml')
+
+k8s_attach(
+  'example-html',
+  'deployment/example-html',
+  live_update=[
+    sync('./', '/src'),
+  ],
+  deps=['./'],
```

Comment:
@milas i wonder if we could auto-populate deps, rather than force it to be specified twice?

Reply:
Yeah, on the extension side, it'd be nice if the LU Starlark objects were structs so you could read `.source_path` off of them to build up a deps list from the rules (e.g. this would also be handy for `file_sync_only`). Though this probably isn't worth the effort (might be better served by e.g. figuring out how to more deeply integrate the API codegen'd variants)

For `k8s_custom_deploy` specifically, we could also do this on the Go side during assembly - I thought about this originally but wasn't sure if it was worth different semantics than `docker_build` (even though they're _very_ different, their APIs are quite similar). And then `deps` could become `apply_deps=[]` (optional) which makes more sense. WDYT?

Reply:
ya tbh we should make sync an alias of LiveUpdateSync and use the autogenerated objs